core/src/main/scala/kafka/server/metadata/BrokerMetadataPublisher.scala

```diff
@@ -137,6 +139,7 @@ class BrokerMetadataPublisher(
     manifest: LoaderManifest
   ): Unit = {
     val highestOffsetAndEpoch = newImage.highestOffsetAndEpoch()
+    val currentMetadataVersion = metadataCache.metadataVersion()
```

Comment:
There's no need to access metadataCache here. That would incur a performance penalty because you're dealing with volatiles. Instead, you can just look at `delta.image().features().metadataVersion()` (or something like that, I didn't check field names)

`delta.image` is the previous metadata image (that the delta is built on top of)